keanu-project/src/main/java/io/improbable/keanu/vertices/ConstantVertexFactory.java

```diff
@@ -74,4 +74,11 @@ public static ConstantDoubleVertex of(DoubleTensor value) {
     public static <TENSOR extends Tensor> ConstantGenericVertex<TENSOR> of(TENSOR tensor) {
         return new ConstantGenericVertex<>(tensor);
     }
+
+    public static boolean isConstantVertex(Vertex v) {
```

Comment:
How about defining a tag interface and putting it on these classes and then using something like assignableFrom (https://stackoverflow.com/questions/3949260/java-class-isinstance-vs-class-isassignablefrom)

Reply:
Can we discuss this further tomorrow?  I've created an interface, but I couldn't see a good place in the hierarchy to pull together all the constants under a single class umbrella.

Reply:
@gordoncaleb - isn't `ConstantVertex` the tag interface you're asking for?

Reply:
Done